### PR TITLE
#430 - Adjusting code to use `key` column in Mysql database

### PR DESCRIPTION
**Pull request by duttarnab**
#430 - Add support for JDBC connection to be able connect to any RDBMS 
https://github.com/GluuFederation/oxd/issues/430

Adjusting code to use `key` column in Mysql database by enclosing key with ` character for Mysql db

### CODE REVIEW

File: oxd-server/src/main/java/org/gluu/oxd/server/persistence/SqlPersistenceServiceImpl.java [hunk 1]
```diff
@@ -341,4 +345,14 @@ public boolean deleteAllExpiredObjects() {
             IOUtils.closeSilently(conn);
         }
     }
+
+    private static String getKeyColumnName(ConfigurationService configurationService) {
+        String driverClass = configurationService.getConfiguration().getStorageConfiguration().get("driver")!= null ?
+                configurationService.getConfiguration().getStorageConfiguration().get("driver").asText() : null;
+
+        if (!Strings.isNullOrEmpty(driverClass) && driverClass.contains("mysql")) {
+            return "`key`";
```
Comment (yuriyz) [inline oxd-server/src/main/java/org/gluu/oxd/server/persistence/SqlPersistenceServiceImpl.java]: if it's `MySQL` you meant to return `obj_key`, no ? (if `key` is not allowed)

File: oxd-server/src/main/java/org/gluu/oxd/server/persistence/SqlPersistenceServiceImpl.java [hunk 2]
```diff
@@ -341,4 +345,14 @@ public boolean deleteAllExpiredObjects() {
             IOUtils.closeSilently(conn);
         }
     }
+
+    private static String getKeyColumnName(ConfigurationService configurationService) {
+        String driverClass = configurationService.getConfiguration().getStorageConfiguration().get("driver")!= null ?
```
Comment (yuriyz) [inline oxd-server/src/main/java/org/gluu/oxd/server/persistence/SqlPersistenceServiceImpl.java]: additional variable is easier to read
```
JsonNode driverNode = configurationService.getConfiguration().getStorageConfiguration().get("driver")
String driverClass = driverNode != null ? driverNode.asText : null;
```

File: oxd-server/src/main/java/org/gluu/oxd/server/persistence/SqlPersistenceServiceImpl.java [hunk 3]
```diff
@@ -23,10 +25,12 @@
     private static final Logger LOG = LoggerFactory.getLogger(SqlPersistenceServiceImpl.class);
 
     private SqlPersistenceProvider provider;
+    private ConfigurationService configurationService;
 
     @Inject
-    public SqlPersistenceServiceImpl(SqlPersistenceProvider provider) {
+    public SqlPersistenceServiceImpl(SqlPersistenceProvider provider, ConfigurationService configurationService) {
         this.provider = provider;
+        this.configurationService = configurationService;
```
Comment (yuriyz) [inline oxd-server/src/main/java/org/gluu/oxd/server/persistence/SqlPersistenceServiceImpl.java]: `this.expiredObjectColumnName = getKeyColumnName(configurationService);`
Then we calculate it one time during `SqlPersistenceServiceImpl` creation and then re-use in all other places.

File: oxd-server/src/main/java/org/gluu/oxd/server/persistence/SqlPersistenceServiceImpl.java (hunk 1 above)
Comment (duttarnab) [reply oxd-server/src/main/java/org/gluu/oxd/server/persistence/SqlPersistenceServiceImpl.java]: @yuriyz , I found Mysql is allowing `key` column name if it is enclosed by ` character. But other databases (h2, postgres) donot allow ` character in query. So to keep the column name `key` for all databases we are using ` in Mysql database.

File: oxd-server/src/main/java/org/gluu/oxd/server/persistence/SqlPersistenceServiceImpl.java (hunk 2 above)
Comment (duttarnab) [reply oxd-server/src/main/java/org/gluu/oxd/server/persistence/SqlPersistenceServiceImpl.java]: done

File: oxd-server/src/main/java/org/gluu/oxd/server/persistence/SqlPersistenceServiceImpl.java (hunk 3 above)
Comment (duttarnab) [reply oxd-server/src/main/java/org/gluu/oxd/server/persistence/SqlPersistenceServiceImpl.java]: done

**Pull request merged**